hack/generator/pkg/codegen/pipeline_create_arm_types.go

```diff
@@ -340,22 +350,25 @@ func addArmConversionInterface(
 	idFactory astmodel.IdentifierFactory,
 	isResource bool) (astmodel.TypeDefinition, error) {
 
-	armObjectType, ok := armDef.Type().(*astmodel.ObjectType)
-	if !ok {
-		return astmodel.TypeDefinition{}, errors.Errorf("arm def %q was not of type object", armDef.Name())
+	objectType, err := astmodel.TypeAsObjectType(armDef.Type())
+	if err != nil {
+		nilDef := astmodel.TypeDefinition{}
+		return nilDef, errors.Errorf("ARM definition %q did not define an object type", armDef.Name())
 	}
 
 	addInterfaceHandler := func(t *astmodel.ObjectType) (*astmodel.ObjectType, error) {
 		return t.WithInterface(armconversion.NewArmTransformerImpl(
 			armDef.Name(),
-			armObjectType,
+			objectType,
 			idFactory,
 			isResource)), nil
 	}
 
-	return transformTypeDefinition(
+	transformedDef, err := transformTypeDefinition(
```

Comment:
minor: Why assign and then return here rather than just return the result of the call like before?